bayesian/utility/v2/component_analyses.py

```diff
@@ -0,0 +1,153 @@
+        ecosystem, package, version, g.decoded_token)
+    return True
+
+
+def ca_validate_input(input_json: Dict, ecosystem: str) -> List[Dict]:
+    """Validate CA Input."""
+    if not input_json:
+        error_msg = "Expected JSON request"
+        raise BadRequest(error_msg)
+
+    if not isinstance(input_json, dict):
+        error_msg = "Expected list of dependencies in JSON request"
+        raise BadRequest(error_msg)
+
+    if not check_for_accepted_ecosystem(ecosystem):
+        error_msg: str = f"Ecosystem {ecosystem} is not supported for this request"
+        raise BadRequest(error_msg)
+
+    if not input_json.get('package_versions'):
+        error_msg: str = "package_versions is missing"
+        raise BadRequest(error_msg)
+
+    packages_list = []
+    for pkg in input_json.get('package_versions'):
+        package = pkg.get("package")
+        version = pkg.get("version")
+        if not all([package, version]):
+            error_msg = "Invalid Input: Package, Version are required."
+            raise BadRequest(error_msg)
+
+        if (not isinstance(version, str)) or (not isinstance(package, str)):
+            error_msg = "Package version should be string format only."
+            raise BadRequest(error_msg)
+
+        if not validate_version(version):
+            error_msg = "Package version should not have special characters."
+            raise BadRequest(error_msg)
+
+        if ecosystem == 'maven':
+            package = MavenCoordinates.normalize_str(package)
+
+        package = case_sensitivity_transform(ecosystem, package)
+        packages_list.append({"name": package, "version": version})
+
+    return packages_list
+
+
+def get_ca_batch_response(ecosystem: str, packages: List[Dict]) -> Tuple[Dict, Set]:
+    """Fetch analysis for given package+version from the graph database.
+
+    This Function does 2 actions:
+    1. Queries GraphDB to fetch Package info
+    2. Calculates Unknown Packages
+
+    :param ecosystem: Ecosystem.
+    :param packages: List of dict of package, version info.
+
+    :returns: Graph Response, Unknown Packages
+    """
+    logger.debug('Executing CA Batch Vendor Specific Analyses')
+    graph_response: Dict = GraphAnalyses().get_batch_ca_data(ecosystem, packages)
+    analyzed_dependencies: Set = set(analysed_package_details(graph_response))
+    unknown_pkgs: Set = get_all_unknown_packages(analyzed_dependencies, packages)
+    return graph_response, unknown_pkgs
+
+
+def analysed_package_details(graph_response: Dict) -> Set:
+    """Analyses Package Details from GraphDB.
+
+    Converts GraphDb output packages into set of Normalised hashable Packages
+    :param graph_response: Graph DB Response
+    :return: set of hashable Packages
+    """
+    db_pkg_list = []
+    for pack_details in graph_response.get('result').get('data'):
+        pkg_name = pack_details.get('package').get('name', [''])[0]
+        pkg_vr = pack_details.get('version').get('version', [''])[0]
```

Comment:
Instead of being permissive, you assert or fail explicitly incase if response doesn't contain `name` & `version` fields.

Reply:
That's why parent component, caller function in `api_v2.py` is in try: catch
this can handle all such errors.